core/src/main/java/com/linecorp/armeria/server/HttpResponseSubscriber.java

```diff
@@ -53,10 +53,15 @@ void onResponseHeaders(ResponseHeaders headers) {
         } else {
             if (req.method() == HttpMethod.HEAD) {
                 endOfStream = true;
-            } else if (status.isContentAlwaysEmpty()) {
-                setState(State.NEEDS_TRAILERS);
             } else {
-                setState(State.NEEDS_DATA_OR_TRAILERS);
+                if (!reqCtx.additionalResponseTrailers().isEmpty()) {
```

Comment:
Note: I haven't looked into the details, but I think the `HttpResponse` itself still can have trailers.
I'm a little concerned that 
1. `headers.isEndOfStream() == true`
2. `reqCtx.additionalResponseTrailers()` is empty
3. but the actual `HttpResponse` has trailers
4. and `setDone(true);` is invoked

I'm not sure why this part of the code changed, but just wanted to make sure there is no issue in this scenario

Reply:
`headers.isEndOfStream()` is not set to `true` if an `HttpResponse` has trailers.
The value is set to true only when the factory methods of `HttpResponse` statically know that the content and trailers are empty.

One possible case is that a user reuses the headers with EOS to create an `HttpResponse` containing body or tailers. It would not be a common case and the fact that `headers` was EOS when creating the original `HttpResponse` is true, I didn't think we need to consider the case.